### PR TITLE
raftstore: getter/setter for region

### PR DESCRIPTION
**Pull request by overvenus**
Add getter and setter for region, so it is easy to add a hook when we updates Peer's region.

Cc #2639

### CODE REVIEW

File: src/raftstore/store/peer_storage.rs [hunk 1]
```diff
@@ -621,10 +621,14 @@ impl PeerStorage {
         self.apply_state.get_truncated_state().get_term()
     }
 
-    pub fn get_region(&self) -> &metapb::Region {
+    pub fn region(&self) -> &metapb::Region {
         &self.region
     }
 
+    pub fn region_mut(&mut self) -> &mut metapb::Region {
```
Comment (BusyJay) [inline src/raftstore/store/peer_storage.rs]: Why not provide a `set_region` method?

File: src/raftstore/store/peer.rs [hunk 2]
```diff
@@ -518,7 +518,15 @@ impl Peer {
...
-        self.get_store().get_region()
+        self.get_store().region()
+    }
+
+    /// Set the region of a peer.
+    ///
+    /// This will update the region of the peer, caller must ensure the region
+    /// has been preserved in a durable device.
+    pub fn set_region(&mut self, region: metapb::Region) {
+        self.mut_store().set_region(region);
```
Comment (Hoverbear) [inline src/raftstore/store/peer.rs]: This `;` is unnecessary.

File: src/raftstore/store/peer_storage.rs [hunk 3]
```diff
@@ -657,7 +661,7 @@ impl PeerStorage {
             return false;
         }
         let snap_epoch = snap_data.get_region().get_region_epoch();
-        let latest_epoch = self.get_region().get_region_epoch();
+        let latest_epoch = self.region.get_region_epoch();
```
Comment (siddontang) [inline src/raftstore/store/peer_storage.rs]: `region()`?

File: src/raftstore/store/peer.rs [hunk 4]
```diff
@@ -518,7 +518,15 @@ impl Peer {
     }
 
     pub fn region(&self) -> &metapb::Region {
```
Comment (siddontang) [inline src/raftstore/store/peer.rs]: I prefer using inline now.

**Comment by overvenus** [comment]
PTAL

**Comment by hicqu** [comment]
LGTM.

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**